### PR TITLE
edv sync implementation

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: integration-tests/cloud-wallet.test.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+} from '@docknetwork/wallet-sdk-data-store/src/types';
+import {
+  SYNC_MARKER_TYPE,
+  initializeCloudWallet,
+} from '@docknetwork/wallet-sdk-core/src/cloud-wallet';
+import {createDataStore} from '@docknetwork/wallet-sdk-data-store-typeorm/src';
+
+const EDV_URL = process.env.EDV_URL;
+
+const agreementKey = {
```
Comment (cykoder) [inline integration-tests/cloud-wallet.test.ts:17]: would it not be better to generate these keys the same way you will generate them for a wallet user?

File: packages/core/src/cloud-wallet.ts [hunk 2]
```diff
@@ -234,6 +234,7 @@ export async function initializeCloudWallet({
   async function clearEdvDocuments() {
     const allDocs = await storageInterface.find({});
 
+    debugger;
```
Comment (cykoder) [inline packages/core/src/cloud-wallet.ts]: can be removed?

File: packages/data-store-web/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+*.sqlite
+lib/
+tsconfig.build.tsbuildinfo
```
Comment (cykoder) [inline packages/data-store-web/.gitignore:3]: missing new line ending

File: packages/data-store-web/scripts/publish.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+yarn build
+npm publish --access public
+
```
Comment (cykoder) [inline packages/data-store-web/scripts/publish.sh:3]: wont let me comment below but `packages/data-store-web/setup-tests.ts` is an empty file

File: packages/data-store-web/src/entities/document/remove-document.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ */
+export async function removeDocument({
+  dataStore,
+  id,
+  options,
+}: ContextProps & {id: string; options?: any}): Promise<void> {
+  assert(!!id, 'Document id is required');
+
+  logger.debug(`Removing document with id ${id}`);
+  // const db = getDataSource(dataStore);
```
Comment (cykoder) [inline packages/data-store-web/src/entities/document/remove-document.ts:24]: commented code, can remove?

File: packages/data-store-web/src/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (cykoder) [inline packages/data-store-web/src/helpers.ts:1]: empty file?

**Review by cykoder — CHANGES_REQUESTED** [review]
couple comments but also this repo could do with a formatter/formatter check to address some inconsistencies with spacings, quotes etc

File: packages/core/src/cloud-wallet.ts (hunk 2 above)
Comment (maycon-mello) [reply packages/core/src/cloud-wallet.ts]: it was removed already, should be fine now

File: packages/data-store-web/src/entities/document/create-document.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      id: json.id,
+    });
+
+    if (existingDocument) {
+      logger.debug(`Document with id ${json.id} already exists`);
+      throw new Error(`Document with id ${json.id} already exists`);
+    }
+  }
+
+  if (!json.id) {
```
Comment (mike-parkhill) [inline packages/data-store-web/src/entities/document/create-document.ts:45]: what happens if the id is specified but a document with that id already exists? Does it overwrite the old one?

File: packages/data-store-web/src/entities/document/get-documens-by-type.ts
Comment (mike-parkhill) [inline packages/data-store-web/src/entities/document/get-documens-by-type.ts:1]: typo in the file name - missing a `t` in `documens`

File: packages/data-store-web/src/entities/document/create-document.ts (hunk 7 above)
Comment (maycon-mello) [reply packages/data-store-web/src/entities/document/create-document.ts:45]: throwing a document with id ${json.id} already exists

**Review by mike-parkhill — APPROVED** [review]

**Review by cykoder — APPROVED** [review]

**Pull request merged**